## IDENTIFIKASI MASALAH

#### Framework SMART

**Case Study: Meningkatkan Peluang Pekerjaan sebagai Data Analyst dengan menggunakan Metode SMART**

Sebagai lulusan baru dari program Teknik Informatika di Institut Teknologi Sepuluh Nopember, saya memiliki keinginan untuk memulai karir sebagai data analyst. Meskipun telah berupaya mencari pekerjaan selama satu bulan terakhir, namun belum berhasil mendapatkan kesempatan pekerjaan di bidang ini. Dalam usaha meningkatkan peluang saya, saya memutuskan untuk fokus pada peningkatan keterampilan dan pemahaman mendalam di bidang data analyst. Untuk mendapatkan wawasan yang lebih baik, saya akan melakukan eksplorasi terhadap job listing data analyst di Amerika Serikat, sehingga saya dapat memperkuat kemampuan dan pengetahuan saya sesuai dengan tuntutan pasar kerja. Tujuannya adalah agar saya dapat mendapatkan pekerjaan yang sesuai dengan kemampuan dan pengetahuan yang telah saya peroleh selama studi dan praktikum di kampus.

1. **Specific:** Menganalisis dataset pekerjaan terbuka, khususnya **LinkedIn Data Analyst jobs listings**, untuk mendapatkan wawasan tentang tren pekerjaan, karakteristik perusahaan, dan persyaratan kualifikasi.

2. **Measurable:**
    - Menetapkan target gaji di atas 8.5 juta sebagai parameter keberhasilan.
    - Melamar pekerjaan sebagai Data Analyst setidaknya pada 20-30 pekerjaan sebagai langkah konkrit dalam mencapai tujuan dalam waktu 2 minggu.

3. **Achievable:** Dengan memanfaatkan dataset LinkedIn, saya dapat menemukan insight baru yang dapat meningkatkan kemungkinan mendapatkan pekerjaan sebagai data analyst.

4. **Relevant:** Analisis terhadap job listings relevan karena akan memberikan gambaran langsung tentang tuntutan pasar kerja untuk posisi Data Analyst.

5. **Time-Bound:**  Menyelesaikan eksplorasi dan analisis dataset LinkedIn dalam waktu 1 bulan.

**Kesimpulan:**
Dalam upaya meningkatkan peluang pekerjaan sebagai Data Analyst, saya merancang sebuah studi kasus dengan menggunakan metode SMART. Fokus utama studi ini adalah menganalisis dataset pekerjaan terbuka, khususnya LinkedIn Data Analyst jobs listings, untuk memahami tren pekerjaan, karakteristik perusahaan, dan persyaratan kualifikasi. Dengan menetapkan target gaji di atas 8.5 juta dan merencanakan pengiriman 20-30 lamaran pekerjaan dalam waktu 2 minggu, saya berharap dapat meningkatkan peluang saya untuk mendapatkan pekerjaan yang sesuai dengan kemampuan dan pengetahuan yang saya miliki. Analisis terhadap job listings ini diharapkan memberikan wawasan langsung tentang tuntutan pasar kerja, sehingga dapat menjadi landasan yang kuat dalam mempersiapkan diri untuk bersaing di dunia kerja sebagai seorang Data Analyst.

## DATA LOADING

Melakukan eksplorasi data untuk mengetahui data-data yang tidak sesuai dengan keinginan kita

### Memuat Data `linkedin jobs`

In [3]:
# Memuat data
import pandas as pd

df1 = pd.read_csv('dataset/linkedin-jobs-africa.csv')
df2 = pd.read_csv('dataset/linkedin-jobs-canada.csv')
df3 = pd.read_csv('dataset/linkedin-jobs-usa.csv')

df = pd.concat([df1, df2, df3], ignore_index=True)
df



title                    company  \
0                   Data Analyst                        NNC   
1                   Data Analyst                Watu Credit   
2     Data Analyst (Credit Risk)                       Kuda   
3                   Data Analyst                Watu Credit   
4                   Data Analyst  Tata Consultancy Services   
...                          ...                        ...   
8485         Junior Data Analyst         Iris Software Inc.   
8486          Data Analyst (SQL)              Marwood Group   
8487                Data Analyst         SmartSense by Digi   
8488                Data Analyst             Synergy Search   
8489                Data Analyst             Insight Global   

                                            description onsite_remote  \
0     We are looking for a passionate certified Data...        onsite   
1     WATU is a focus-driven, customer-centric compa...        onsite   
2     Kuda is a fintech on a mission to make financi...        onsite   
3     Role: Data AnalystReporting to: Data Analytics...        onsite   
4     About Tata Consultancy ServicesTata Consultanc...        onsite   
...                                                 ...           ...   
8485  Iris's client, one of the world's largest fina...        hybrid   
8486  The Marwood Group (Marwood) is a leading healt...        hybrid   
8487  Join a high-performing, tight-knit team at a f...        hybrid   
8488  Nashville (Berry Hill) based company looking t...        hybrid   
8489  Overview - Minimum 5 years in a quantitative d...        hybrid   

                                                 salary  \
0                                                   NaN   
1                                                   NaN   
2                                                   NaN   
3                                                   NaN   
4                                                   NaN   
...                                                 ...   
8485                                                NaN   
8486  $75,000.00\r\n            -\r\n            $95...   
8487                                                NaN   
8488  $85,000.00\r\n            -\r\n            $95...   
8489  $130,000.00\r\n            -\r\n            $1...   

                                         location  \
0                             Cairo, Cairo, Egypt   
1                           Nairobi County, Kenya   
2           Cape Town, Western Cape, South Africa   
3                  Nairobi, Nairobi County, Kenya   
4     City of Johannesburg, Gauteng, South Africa   
...                                           ...   
8485                         Texas, United States   
8486              New York City Metropolitan Area   
8487                                Mishawaka, IN   
8488                  Nashville Metropolitan Area   
8489                                  Burbank, CA   

                                               criteria posted_date  \
0     [{'مستوى الأقدمية': 'غير مطبق'}, {'نوع التوظيف...  2022-11-18   
1     [{'Seniority level': 'Associate'}, {'Employmen...  2022-11-01   
2     [{'Seniority level': 'Not Applicable'}, {'Empl...  2022-11-21   
3     [{'Seniority level': 'Mid-Senior level'}, {'Em...  2022-11-01   
4     [{'Seniority level': 'Mid-Senior level'}, {'Em...  2022-10-27   
...                                                 ...         ...   
8485  [{'Seniority level': 'Mid-Senior level'}, {'Em...  2022-10-21   
8486  [{'Seniority level': 'Mid-Senior level'}, {'Em...  2022-10-26   
8487  [{'Seniority level': 'Associate'}, {'Employmen...  2022-11-17   
8488  [{'Seniority level': 'Not Applicable'}, {'Empl...  2022-11-14   
8489  [{'Seniority level': 'Mid-Senior level'}, {'Em...  2022-11-14   

                                                   link  
0     https://eg.linkedin.com/jobs/view/data-analyst...  
1     https://ke.linkedin.com/jobs/view/data-analyst...  
2     https://za.lin

### Melihat informasi data lebih detail

In [20]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8490 entries, 0 to 8489
Data columns (total 9 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   title          8490 non-null   object
 1   company        8490 non-null   object
 2   description    8490 non-null   object
 3   onsite_remote  8490 non-null   object
 4   salary         965 non-null    object
 5   location       8490 non-null   object
 6   criteria       8490 non-null   object
 7   posted_date    8490 non-null   object
 8   link           8490 non-null   object
dtypes: object(9)
memory usage: 597.1+ KB


Informasi lain:
1. Data berjumlah `8490`
2. Memiliki kolom sebanyak `9` kolom
3. Teindikasi ada data yang `Nan`
4. Pada kolom `criteria` akan dilakukan pemecahan isi data untuk dibagi lagi menjadi kolom baru
5. Mengubah `posted_date` menjadi type data `to_datetime`

In [21]:
df['posted_date']=pd.to_datetime(df['posted_date'])
df['posted_date'].dtypes

dtype('<M8[ns]')

## Data Cleaning

In [22]:
df.duplicated().sum()

0

In [23]:
df['salary'].value_counts()

salary
$50.00\r\n            -\r\n            $60.00              89
$100,000.00\r\n            -\r\n            $120,000.00    50
$85,000.00\r\n            -\r\n            $100,000.00     38
$125,000.00\r\n            -\r\n            $155,000.00    37
$60.00\r\n            -\r\n            $60.00              37
$40.00\r\n            -\r\n            $60.00              37
$30.00\r\n            -\r\n            $33.00              37
$120,000.00\r\n            -\r\n            $130,000.00    36
$30.00\r\n            -\r\n            $30.00              36
$59.69\r\n            -\r\n            $79.58              36
$22.00\r\n            -\r\n            $24.00              35
$80,000.00\r\n            -\r\n            $90,000.00      35
$73,036.00\r\n            -\r\n            $105,901.00     35
$135,000.00\r\n            -\r\n            $140,000.00    34
$70,000.00\r\n            -\r\n            $80,000.00      34
$60,000.00\r\n            -\r\n            $90,000.00      34
$

In [24]:
df['salary'] = df['salary'].str.replace('\r\n', '').str.replace(',', '').str.strip()


In [25]:
df.value_counts('salary')

salary
$50.00            -            $60.00            89
$100000.00            -            $120000.00    50
$85000.00            -            $100000.00     38
$30.00            -            $33.00            37
$40.00            -            $60.00            37
$125000.00            -            $155000.00    37
$60.00            -            $60.00            37
$120000.00            -            $130000.00    36
$59.69            -            $79.58            36
$30.00            -            $30.00            36
$80000.00            -            $90000.00      35
$73036.00            -            $105901.00     35
$22.00            -            $24.00            35
$135000.00            -            $140000.00    34
$70000.00            -            $80000.00      34
$60000.00            -            $90000.00      34
$40.00            -            $50.00            34
$80000.00            -            $110000.00     34
$120000.00            -            $135000.00    34
$9000

In [26]:
df.isnull().sum()

title               0
company             0
description         0
onsite_remote       0
salary           7525
location            0
criteria            0
posted_date         0
link                0
dtype: int64

In [27]:
df_copy = df.dropna()

In [28]:
df_copy.info()

<class 'pandas.core.frame.DataFrame'>
Index: 965 entries, 3849 to 8489
Data columns (total 9 columns):
 #   Column         Non-Null Count  Dtype         
---  ------         --------------  -----         
 0   title          965 non-null    object        
 1   company        965 non-null    object        
 2   description    965 non-null    object        
 3   onsite_remote  965 non-null    object        
 4   salary         965 non-null    object        
 5   location       965 non-null    object        
 6   criteria       965 non-null    object        
 7   posted_date    965 non-null    datetime64[ns]
 8   link           965 non-null    object        
dtypes: datetime64[ns](1), object(8)
memory usage: 75.4+ KB


## Exploratory Data Analysis (EDA)

### Informasi umum pekerjaan sebagai Data Analyst

#### Top 10 Perusahaan yang membuka lowongan terbanyak

In [66]:
import plotly.express as px

# Menghitung jumlah lowongan pekerjaan untuk setiap perusahaan
company_counts = df_copy['company'].value_counts()

# Mengambil top 10 perusahaan
top_companies = company_counts.nlargest(10).index

# Filter DataFrame hanya untuk top 10 perusahaan
df_top_companies = df_copy[df_copy['company'].isin(top_companies)]

# Menambahkan kolom baru 'company_counts' yang berisi jumlah lowongan perusahaan
df_top_companies['company_counts'] = df_top_companies['company'].map(company_counts)

# Mengurutkan DataFrame berdasarkan jumlah lowongan secara menurun
df_top_companies = df_top_companies.sort_values(by='company_counts', ascending=False)

# Membuat bar plot interaktif dengan efek hover
fig = px.bar(df_top_companies, y='company', color='company', title='Top 10 Company yang membuka lowongan data analyst',
             labels={'company': 'Company'},
             height=600,
             text='company_counts')  # Menambahkan parameter text untuk menampilkan jumlah perusahaan pada hover

# Menampilkan plot
fig.show()


Dari visualisasi di atas didapatkan bahwa perusahaan yang membuka lowongan pekerjaan terbanyak sebagai `Data Analyst` adalah `Insight Global` dengan jumlah `96`, sedangkan perusahaan yang lain rata-rata berjumlah `37`. 

#### Top 10 Daerah/Kota yang membuka lowongan terbanyak

In [71]:
import plotly.express as px
import pandas as pd

# Menghitung jumlah lowongan pekerjaan untuk setiap lokasi
location_counts = df_copy['location'].value_counts()

# Membuat DataFrame baru dari value_counts
df_location_counts = pd.DataFrame({'Lokasi': location_counts.index, 'Jumlah Lowongan': location_counts.values})

# Mengambil top 10 lokasi berdasarkan jumlah lowongan
top_10_locations = df_location_counts.nlargest(10, 'Jumlah Lowongan')

# Membuat pie chart untuk top 10 lokasi dengan jumlah lowongan terbanyak
fig = px.pie(top_10_locations, names='Lokasi', values='Jumlah Lowongan',
             title='Persentase Jumlah Lowongan Data Analyst Berdasarkan Lokasi',
             labels={'Lokasi': 'Lokasi', 'Jumlah Lowongan': 'Jumlah Lowongan'},
             height=600,
             )

# Menampilkan plot
fig.show()


Lowongan pekerjaan sebagai `Data Analyst` banyak dibuka di negara `Amerika Serikat`, jika dilihat dari data lebih dari `50%`.

#### Type job Data Analyst

In [87]:
import plotly.express as px
import pandas as pd
import warnings
# Mengabaikan semua peringatan
warnings.filterwarnings("ignore")
df_copy.rename(columns={'onsite_remote':'job_type'}, inplace=True)

# Hitung jumlah setiap kategori
job_type = df_copy['job_type'].value_counts().rename_axis('job_type').reset_index(name='count')

# Membuat pie chart dengan efek hover menggunakan plotly
fig = px.pie(job_type, values='count', names='job_type',
             title='Distribusi Onsite vs Remote',
             color_discrete_sequence=px.colors.qualitative.Set3,
             hover_data=['count'],
             width=600,# Menambahkan efek hover untuk menampilkan jumlah pada hover
             labels={'count': 'Jumlah'})  # Mengganti label count menjadi Jumlah

# Menampilkan plot
fig.show()


Tipe pekerjaan yang ditawarkan oleh perusahaan lebih banyak `remote` daripada `onsite` maupun `hybrid`.

#### Posting Lowongan Data Analyst

In [76]:
import plotly.express as px
import pandas as pd

# Memastikan bahwa kolom 'posted_date' berisi tipe data datetime
if pd.api.types.is_datetime64_any_dtype(df_copy['posted_date']):
    # Mendapatkan nama hari dari setiap tanggal posting
    days = df_copy['posted_date'].dt.day_name().value_counts().reset_index()
    days.columns = ['Hari', 'Jumlah Posting']

    # Mendefinisikan urutan hari dari Senin hingga Minggu
    days_order = ['Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday', 'Sunday']

    # Membuat bar plot untuk distribusi posting per hari
    fig = px.bar(days, x='Hari', y='Jumlah Posting',
                 labels={'Hari': 'Hari', 'Jumlah Posting': 'Jumlah Posting'},
                 title='Distribusi Jumlah Posting per Hari',
                 height=600, width=1100, color='Jumlah Posting',
                 category_orders={'Hari': days_order})  # Menambahkan urutan kategori

    # Menampilkan nilai di atas bar
    fig.update_traces(text=days['Jumlah Posting'], textposition='outside')

    # Menampilkan plot
    fig.show()
else:
    print("Kolom 'posted_date' tidak dapat diidentifikasi sebagai tipe data datetime.")


Postingan lowongan pekerjaan sebagai `Data Analyst` kebanyakan di post oleh perusahaan pada hari `senin - rabu` dimana jumlahnya lebih dari `190` lowongan.

### Kota dengan gaji terbesar seorang data analyst

In [30]:
import plotly.express as px

# Membuat scatter plot untuk kota dengan gaji terbesar
fig = px.scatter(df_copy, x='salary', y='location', color='salary' ,title='Kota dengan Gaji Tertinggi untuk Data Analyst',
                 labels={'salary': 'Gaji', 'location': 'Kota'},
                 height=600)

# Menampilkan plot
fig.show()


Jika dilihat dari data diatas gaji minimal sebagai seorang `Data Analyst` yaitu dari `$22.00` sampai `$140000.00`.

### Kriteria

In [ ]:
c=df_copy['criteria'].apply(lambda x: x[1:-1].split(',',3)).to_list()

print(c[0]) #list of items as string
c[0][0] #dictionary as string
sl=[]
et=[]
jf=[]
ind=[]
import json
def dict_con(d):
    try:
        d1=(json.loads(i.replace("'",'"')) for i in d)
        for i in d1:
            if "Seniority level" in i.keys():
                sl.extend(i.values())
            if "Employment type" in i.keys():
                et.extend(i.values())
            if "Job function" in i.keys():
                jf.extend(i.values())
            if "Industries" in i.keys():
                ind.extend(i.values())
    except:
        pass


for i in c:
    dict_con(i)

In [ ]:
Seniority_level=pd.Series(sl)
Employment_type=pd.Series(et)
Job_function=pd.Series(jf)
Industries=pd.Series(ind)

#### Pengalaman sebagai Data Analyst

In [79]:
import plotly.express as px
import pandas as pd

# Menghitung jumlah seniority level
Seniority_level = pd.Series(sl)
seniority_counts = Seniority_level.value_counts()

# Membuat pie chart
fig = px.pie(names=seniority_counts.index, values=seniority_counts.values,
             title="Distribusi Seniority Level pada Pekerjaan Data Analyst",
             labels={'names': 'Seniority Level', 'values': 'Number of Jobs'},
             color_discrete_sequence=px.colors.qualitative.Set3,
             height=400, width=600)

# Menampilkan plot
fig.show()


Lowongan sebagai `Data Analyst` mayoritas berpengelaman dengan presentase `44%` jika dibandingkan dengan jumlah dari `entry level` dan `not Applicable` hanya `20%`.

#### Kontrak pekerjaan yang di tawarkan

In [80]:
import plotly.express as px
import pandas as pd

# Menghitung jumlah employment type
Employment_type = pd.Series(et)
employment_counts = Employment_type.value_counts()

# Membuat pie chart
fig = px.pie(names=employment_counts.index, values=employment_counts.values,
             title="Distribusi Employment Types pada Pekerjaan Data Analyst",
             labels={'names': 'Employment Type', 'values': 'Number of Jobs'},
             color_discrete_sequence=px.colors.qualitative.Set3,
             height=400, width=600)

# Menampilkan plot
fig.show()


Pekerjaan sebagai `Data Analyst` seimbang untuk `contract` maupun `Full-time`

#### Lingkup pekerjaan yang akan dilakukan

In [63]:
Job_function=pd.Series(jf)

# Using Plotly Express for bar chart
fig = px.bar(pd.DataFrame({'Job_function': Job_function.value_counts().index,
                           'count': Job_function.value_counts().values}),
             x='count', y='Job_function',
             orientation='h',
             color='Job_function',
             title="Data Analyst Jobs Based On Job_function Expectations",
             labels={'Job_function': 'Job_function', 'count': 'Number of Jobs'},
             height=400, width=1100)

fig.update_layout(yaxis=dict(title='Job_function'),
                  xaxis=dict(title='Number of Jobs'),
                  showlegend=False)

fig.update_traces(text=Job_function.value_counts().values, textposition='outside')

fig.show()


Kebanyakan role pekerjaan sebagai `Data Analyst` bekerja di lingkup `Technology` dan `Finance`.

#### Industri yang menyediakan pekerjaan Data Analyst

In [86]:
Industries=pd.Series(ind)

# Using Plotly Express for bar chart
fig = px.bar(pd.DataFrame({'Industries': Industries.value_counts().index,
                           'count': Industries.value_counts().values}),
             x='count', y='Industries',
             orientation='h',
             color='Industries',
             title="Data Analyst Jobs Based On Industries Expectations",
             labels={'Industries': 'Industries', 'count': 'Number of Jobs'},
             height=400, width=1100)

fig.update_layout(yaxis=dict(title='Industries'),
                  xaxis=dict(title='Number of Jobs'),
                  showlegend=False)

fig.update_traces(text=Industries.value_counts().values, textposition='outside')

fig.show()

Industri yang sering membuka lowongan sebagai `Data Analyst` yaitu industri `IT Services` dan `IT Consulting`.

## Kesimpulan

Dari analisis data, terlihat bahwa Insight Global menonjol sebagai perusahaan dengan aktivitas perekrutan Data Analyst yang tinggi, menciptakan peluang bagi pelamar yang tertarik dengan perusahaan tersebut. Lebih lanjut, Amerika Serikat menjadi pasar pekerjaan utama untuk peran ini, sehingga mengeksplorasi peluang di sana mungkin memberikan keuntungan lebih besar.

Penting untuk diperhatikan bahwa sebagian besar pekerjaan Data Analyst dapat dilakukan secara remote, menawarkan fleksibilitas yang penting bagi banyak profesional. Selain itu, pengumuman lowongan cenderung dilakukan pada awal pekan, memberikan waktu bagi pelamar untuk bersiap dan merespons dengan cepat.

Dalam hal persyaratan, walaupun pengalaman dihargai (44%), peluang masih terbuka lebar untuk pemula (20%) dan bagi mereka yang menganggap pengalaman tidak berlaku (20%). Oleh karena itu, tidak ada salahnya untuk melamar meskipun Anda mungkin berada di awal karir atau sedang mencari perubahan.

Gaji yang bervariasi menunjukkan keragaman dalam penawaran, sehingga sebaiknya calon pelamar melakukan penelitian tambahan untuk menentukan nilai gaji yang wajar berdasarkan kualifikasi dan pengalaman mereka. Selain itu, kerja kontrak dan full-time memiliki porsi yang seimbang, memberikan opsi lebih untuk memenuhi preferensi gaya kerja.

Dengan memahami dominasi industri seperti Technology dan Finance, pelamar dapat mengarahkan pencarian mereka dengan lebih strategis. Khususnya, industri IT Services dan IT Consulting menonjol sebagai sumber peluang yang signifikan.

Secara keseluruhan, melalui analisis ini, calon pelamar Data Analyst dapat mengidentifikasi tren utama, menyesuaikan diri dengan ekspektasi pasar, dan menyiapkan strategi yang tepat untuk mendapatkan peluang pekerjaan yang sesuai dengan keinginan dan kualifikasi mereka.